modules/Calls/Call.php

```diff
@@ -174,7 +174,7 @@ public function save($check_notify = false)
         global $timedate;
 
         if (!empty($this->date_start)) {
-            if (!empty($this->duration_hours) && !empty($this->duration_minutes)) {
+            if (!empty($this->duration_hours + $this->duration_minutes)) {
```

Comment:
Hi @SinergiaCRM,

Thank you for your PR.

I was thinking that maybe something like the following would be safer, what do you think?

```php
$durationHours = '';
if(!is_numeric($this->duration_hours ?? '')) {
    $durationHours = $this->duration_hours;
}

$durationMinutes = '';
if(!is_numeric($$this->duration_minutes ?? '')) {
    $durationMinutes = $this->duration_minutes;
}

if (durationHours !== '' || $durationMinutes !== '') {
    $td = $timedate->fromDb($this->date_start);
    if ($td) {
        $modifyString = "+";
        if(durationHours !== '') {
            $modifyString .= "$durationHours hours ";
        }

        if($durationMinutes !== '') {
            $modifyString .= "$durationMinutes mins ";
        }

        $this->date_end = $td->modify($modifyString)->asDb();
    }
} else {
    $this->date_end = $this->date_start;
}
```

Reply:
Hello @clemente-raposo, I think it can be solved without needing to write such a long code. 
The proposal from @QuickCRM seems quite simple, consistent, and safe. All the tests carried out have worked correctly. I believe that is what we should implement.